### PR TITLE
Typo fixes

### PR DESCRIPTION
**Pull request by jnm2**
I spotted a couple and ran ReSpeller over the solution. For a codebase this big, there are hardly any typos!

The XML doc comments and exception message typos are the most important, but if someone is willing to put an eye on all these changes that would be great.

### CODE REVIEW

File: src/NUnitFramework/framework/Assert.cs [hunk 1]
```diff
@@ -374,7 +374,7 @@ private static void IssueWarning(string message)
             result.RecordAssertion(AssertionStatus.Warning, message, GetStackTrace());
         }
 
-        // System.Envionment.StackTrace puts extra entries on top of the stack, at least in some environments
+        // System.Environment.StackTrace puts extra entries on top of the stack, at least in some environments
```
Comment (jnm2) [inline src/NUnitFramework/framework/Assert.cs]: I'd really like to use `<see cref="System.Environment.StackTrace"/>` here but that should probably be its own PR and take care of more than just the typo spots.

File: src/NUnitFramework/framework/Api/FrameworkPackageSettings.cs [hunk 2]
```diff
@@ -52,7 +52,7 @@ public static class FrameworkPackageSettings
 
         /// <summary>
         /// Full path of the directory to be used for work and result files.
-        /// This path is provided to tests by the frameowrk TestContext.
+        /// This path is provided to tests by the framework TestContext.
```
Comment (jnm2) [inline src/NUnitFramework/framework/Api/FrameworkPackageSettings.cs]: Same here with `<see cref="TestContext"/>`. Gives really nice IntelliSense coloring and help linking.

**Comment by CharliePoole** [comment]
These all look good to merge but I think it should wait till after the release and merging back of the release branch.

File: src/NUnitFramework/framework/SpecialValue.cs [hunk 3]
```diff
@@ -33,7 +33,7 @@ public enum SpecialValue
     {
         /// <summary>
         /// Null represents a null value, which cannot be used as an 
-        /// argument to an attriute under .NET 1.x
+        /// argument to an attribute under .NET 1.x
```
Comment (ChrisMaddock) [inline src/NUnitFramework/framework/SpecialValue.cs]: Under .NET 1.X 😱 @CharliePoole - Do you think we can get rid of this now, or does it serve other uses?

Comment (CharliePoole) [inline src/NUnitFramework/framework/SpecialValue.cs]: My guess is that we can probably get rid of SpecialValue entirely, but since this PR was about typos I didn't want to confuse the issue.

**Comment by CharliePoole** [comment]
Looks good to merge for me as well. @rprouse Do you want us to keep merging to master or will that confuse the release process at this point?

File: src/NUnitFramework/framework/SpecialValue.cs (hunk 3 above)
Comment (ChrisMaddock) [inline src/NUnitFramework/framework/SpecialValue.cs]: I've made https://github.com/nunit/nunit/issues/1963 😄

Comment (ChrisMaddock) [inline src/NUnitFramework/framework/SpecialValue.cs]: By the way, @jnm2 - I forgot to mention how happy this PR makes my inner-perfectionist. 😄 Nice one!

**Comment by rprouse** [comment]
I'm good to merge this. I am going to do another merge from master into the release branch so may as well get this in.
